juju/juju
# [JUJU-2309] Add confirmation prompt to remove-unit cmd

Implement the dry-run option for DestroyUnit, and use this to generate a preview in the confirmation prompt

Also do some minor cleanups as a flyby

## Checklist


## QA steps

With both a new and old (i.e. the current release) controller, verify the following commands act as expected

- `juju remove-unit` & `juju remove-unit --no-prompt` should behave as before. No prompts should be shown
- `juju remove-unit --dry-run` should output a readable dry run of what will be removed for a new controller. It should fail smoothly for an old controller
- `JUJU_SKIP_CONFIRMATION=0 juju remove-unit` should prompt the user before continuing. For a new controller, a dry-run should be provided as well. 

For extra credit try removing a unit with storage attached
